xds/src/main/java/io/grpc/xds/sds/SslContextSecretVolumeSecretProvider.java

```diff
@@ -99,28 +105,41 @@ static TlsCertificate validateTlsCertificate(
   }
 
   static SslContextSecretVolumeSecretProvider getProviderForServer(
-      TlsCertificate tlsCertificate, @Nullable CertificateValidationContext certContext) {
+      DownstreamTlsContext downstreamTlsContext) {
+    checkNotNull(downstreamTlsContext, "downstreamTlsContext");
+    CommonTlsContext commonTlsContext = downstreamTlsContext.getCommonTlsContext();
+    TlsCertificate tlsCertificate = null;
+    if (commonTlsContext.getTlsCertificatesCount() > 0) {
+      tlsCertificate = commonTlsContext.getTlsCertificates(0);
+    }
     // first validate
     validateTlsCertificate(tlsCertificate, /* optional= */ false);
+    CertificateValidationContext certContext = commonTlsContext.getValidationContext();
```

Comment:
this is also oneof field, how we make sure this exists? do we need to handle other 2 types?
also, the variable name is misleading, certification *validation* context is not certification context.

Reply:
Good points. I will address both of them. Regarding oneof I thought the control plane will not send inconsistent data (one type in `TlsCertitificate` and another type for `validationContext`) but it is good to put the checks in.

Reply:
Fixed.